tsl/src/compression/compression.c

```diff
@@ -707,10 +707,13 @@ static void
 row_compressor_update_group(RowCompressor *row_compressor, TupleTableSlot *row)
 {
 	int col;
-
+	/* save original memory context */
+	const MemoryContext oldcontext = CurrentMemoryContext;
 	Assert(row_compressor->rows_compressed_into_current_value == 0);
 	Assert(row_compressor->n_input_columns <= row->tts_nvalid);
 
+	if (row_compressor->n_input_columns)
+		MemoryContextSwitchTo(row_compressor->per_row_ctx);
```

Comment:
```suggestion
		MemoryContextSwitchTo(row_compressor->per_row_ctx->parent);
```

Reply:
I don't know if it was wrong or what, but this would be the equivalent.

Reply:
I dont know, there were coredumps in CI tests. Now i changed as you suggest. Let me see if CI checks passes.